### PR TITLE
Permissions audit

### PR DESCRIPTION
**Pull request by kamorel**
# Description

Permissions audit based on document provided in SHOWCASE-3148.

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: frontend/src/store/permissionStore.ts [hunk 1]
```diff
@@ -257,8 +268,11 @@ export const usePermissionStore = defineStore('permission', () => {
   async function mapObjectToUserPermissions(objectId: string) {
     try {
       appStore.beginIndeterminateLoading();
+      console.log(objectId);
       const objectPerms = state.objectPermissions.value.filter((x: COMSObjectPermission) => x.objectId === objectId);
+      console.log(objectPerms);
```
Comment (dc-bcgov) [inline frontend/src/store/permissionStore.ts:273]: Is this on purpose?

File: frontend/src/store/permissionStore.ts [hunk 2]
```diff
@@ -257,8 +268,11 @@ export const usePermissionStore = defineStore('permission', () => {
   async function mapObjectToUserPermissions(objectId: string) {
     try {
       appStore.beginIndeterminateLoading();
+      console.log(objectId);
```
Comment (dc-bcgov) [inline frontend/src/store/permissionStore.ts:271]: Is this on purpose?

**Review by dc-bcgov — COMMENTED** [review]
Left three comments. Love the work you've done here.

File: frontend/src/store/permissionStore.ts [hunk 3]
```diff
@@ -257,8 +268,11 @@ export const usePermissionStore = defineStore('permission', () => {
...
       const objectPerms = state.objectPermissions.value.filter((x: COMSObjectPermission) => x.objectId === objectId);
+      console.log(objectPerms);
       const uniqueIds = [...new Set(objectPerms.map((x: COMSObjectPermission) => x.userId))];
+      console.log(uniqueIds);
```
Comment (dc-bcgov) [inline frontend/src/store/permissionStore.ts:275]: Is this on purpose?

File: frontend/src/components/object/ObjectTable.vue [hunk 4]
```diff
@@ -178,13 +199,11 @@ watch( selectedObjects, () => {
...
         <template #body="{ data }">
           <ShareObjectButton
-            v-if="permissionStore.isObjectActionAllowed(
-              data.id, getUserId, Permissions.READ, props.bucketId as string)"
             :id="data.id"
           />
           <DownloadObjectButton
             v-if="permissionStore.isObjectActionAllowed(
-              data.id, getUserId, Permissions.READ, props.bucketId as string)"
+              data.id, getUserId, Permissions.READ, props.bucketId as string) || data.public"
```
Comment (jujaga) [inline frontend/src/components/object/ObjectTable.vue:206]: It's better to evaluate obj.public first to leverage short circuiting; that way we can save some computation if it's already public and avoid needing to hit the permissionStore.

File: frontend/src/components/object/ObjectTable.vue [hunk 5]
```diff
@@ -197,6 +216,8 @@ watch( selectedObjects, () => {
             <font-awesome-icon icon="fa-solid fa-users" />
           </Button>
           <Button
+            v-if="permissionStore.isObjectActionAllowed(
+              data.id, getUserId, Permissions.READ, props.bucketId as string) || data.public"
```
Comment (jujaga) [inline frontend/src/components/object/ObjectTable.vue:220]: It's better to evaluate obj.public first to leverage short circuiting; that way we can save some computation if it's already public and avoid needing to hit the permissionStore.

**Review by jujaga — COMMENTED** [review]
Semantically looks pretty good. A couple relatively minor considerations to make.

**Review by dc-bcgov — APPROVED** [review]

**Review by jujaga — APPROVED** [review]

**Pull request merged**